[BUG] Issues with fake players and plugins.

**Describe the bug**
A clear and concise description of what the bug is:
A "fake player" throws errors on Residence plugin

**To Reproduce**
Steps to reproduce the behavior:
1. Place Farming Station
2. Configure Farming Station start them
3. See error

**Expected behavior**
A clear and concise description of what you expected to happen:
Not to error?

**Screenshots**
If applicable, add screenshots to help explain your problem:

Magma.yml tried to add Fake Players permissions
[https://ctrlv.cz/teAu](url)


Error:
> [Server thread/ERROR] [Minecraft]: Could not pass event BlockBreakEvent to Residence v4.8.8.2
org.bukkit.event.EventException: null
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:68) ~[EventExecutor.class:761933c-STABLE]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[TimedEventExecutor.class:761933c-STABLE]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[RegisteredListener.class:761933c-STABLE]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:517) [SimplePluginManager.class:761933c-STABLE]
	at org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callBlockBreakEvent(CraftEventFactory.java:1234) [CraftEventFactory.class:761933c-STABLE]
	at net.minecraftforge.event.world.BlockEvent$HarvestDropsEvent.<init>(BlockEvent.java:113) [BlockEvent$HarvestDropsEvent.class:761933c-STABLE]
	at net.minecraftforge.event.ForgeEventFactory.fireBlockHarvesting(ForgeEventFactory.java:343) [ForgeEventFactory.class:761933c-STABLE]
	at crazypants.enderio.base.farming.farmers.StemFarmer.harvestBlock(StemFarmer.java:71) [StemFarmer.class:?]
	at crazypants.enderio.base.farming.registry.Commune$2.run(Commune.java:44) [Commune$2.class:?]
	at crazypants.enderio.base.farming.registry.Commune$2.run(Commune.java:41) [Commune$2.class:?]
	at crazypants.enderio.base.farming.registry.Registry.foreach(Registry.java:36) [Registry.class:?]
	at crazypants.enderio.base.farming.registry.Commune.harvestBlock(Commune.java:41) [Commune.class:?]
	at crazypants.enderio.machines.machine.farm.TileFarmStation.executeHarvest(TileFarmStation.java:342) [TileFarmStation.class:?]
	at crazypants.enderio.machines.machine.farm.TileFarmStation.doTick(TileFarmStation.java:297) [TileFarmStation.class:?]
	at crazypants.enderio.machines.machine.farm.TileFarmStation.checkProgress(TileFarmStation.java:245) [TileFarmStation.class:?]
	at crazypants.enderio.base.machine.baselegacy.AbstractPoweredTaskEntity.processTasks(AbstractPoweredTaskEntity.java:90) [AbstractPoweredTaskEntity.class:?]
	at crazypants.enderio.base.machine.base.te.AbstractMachineEntity.doUpdate(AbstractMachineEntity.java:256) [AbstractMachineEntity.class:?]
	at crazypants.enderio.base.machine.baselegacy.AbstractPoweredMachineEntity.doUpdate(AbstractPoweredMachineEntity.java:62) [AbstractPoweredMachineEntity.class:?]
	at crazypants.enderio.machines.machine.farm.TileFarmStation.doUpdate(TileFarmStation.java:223) [TileFarmStation.class:?]
	at com.enderio.core.common.TileEntityBase.func_73660_a(TileEntityBase.java:49) [TileEntityBase.class:?]
	at net.minecraft.world.World.func_72939_s(World.java:2160) [amu.class:?]
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:852) [oo.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:988) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:461) [nz.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:865) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:737) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
Caused by: java.lang.NullPointerException
	at com.bekvon.bukkit.residence.Residence.isResAdminOn(Residence.java:1213) ~[?:?]
	at com.bekvon.bukkit.residence.listeners.ResidenceBlockListener.onBlockBreak(ResidenceBlockListener.java:197) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor138.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:66) ~[EventExecutor.class:761933c-STABLE]
	... 26 more

**Info (please complete the following information):**
- Mod Name: EnderIO
- Plugin Name: Residence
- Plugin Version: Any Version
 - Mod Version: Any Version
 - Magma Version: Magma-761933c-STABLE-server
 - OS: Linux

**Additional context**
Add any other context about the problem here:

--- a/src/main/java/org/bukkit/craftbukkit/v1_12_R1/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/v1_12_R1/event/CraftEventFactory.java
@@ -1212,13 +1212,12 @@ public class CraftEventFactory {
     }
 
     // Cauldron start
-    public static BlockBreakEvent callBlockBreakEvent(net.minecraft.world.World world, BlockPos pos, IBlockState iBlockState, net.minecraft.entity.player.EntityPlayerMP player) {
+    public static BlockBreakEvent callBlockBreakEvent(net.minecraft.world.World world, BlockPos pos, IBlockState iBlockState, EntityPlayerMP player) {
         org.bukkit.block.Block bukkitBlock = world.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
-        org.bukkit.event.block.BlockBreakEvent blockBreakEvent = new org.bukkit.event.block.BlockBreakEvent(bukkitBlock, Bukkit.getPlayer(player.getUniqueID()));
-        EntityPlayerMP playermp = (EntityPlayerMP) player;
+        org.bukkit.event.block.BlockBreakEvent blockBreakEvent = new org.bukkit.event.block.BlockBreakEvent(bukkitBlock, player.getBukkitEntity());
         net.minecraft.block.Block block = iBlockState.getBlock();
-        if (!(playermp instanceof FakePlayer)) {
-            boolean isSwordNoBreak = playermp.interactionManager.getGameType().isCreative() && !playermp.getHeldItemMainhand().isEmpty() && playermp.getHeldItemMainhand().getItem() instanceof ItemSword;
+        if (!(player instanceof FakePlayer)) {
+            boolean isSwordNoBreak = player.interactionManager.getGameType().isCreative() && !player.getHeldItemMainhand().isEmpty() && player.getHeldItemMainhand().getItem() instanceof ItemSword;
             if (!isSwordNoBreak) {
                 int exp = 0;
                 if (!(block == null || !player.canHarvestBlock(block.getDefaultState()) || block.canSilkHarvest(world, pos, block.getBlockState().getBaseState(), player) && EnchantmentHelper.getEnchantmentLevel(Enchantments.SILK_TOUCH, player.getHeldItemMainhand()) > 0)) {

